Core/include/Acts/Utilities/AlgebraHelpers.hpp

```diff
@@ -186,10 +188,21 @@ inline ActsMatrix<A::RowsAtCompileTime, B::ColsAtCompileTime> blockedMult(
 /// @return The theta value
 template <typename MatrixType, typename ResultType = MatrixType>
 std::optional<ResultType> safeInverse(const MatrixType& m) noexcept {
+  constexpr int rows = MatrixType::RowsAtCompileTime;
+  constexpr int cols = MatrixType::ColsAtCompileTime;
+
```

Comment:
How about a static assert if the matrix is symmetric?

Reply:
Depends on what we want from the function. I think you meant square, not symmetric?
In the current case it should give us an "un-invertible" if it is not square, which should be good enough?

Reply:
Right I meant square indeed. Will the invertibility check give a compile time error itself? Depending on what they print as a message it might still be nice to have a custom one.

Reply:
I put a static assert because:
- `computeInverseWithCheck()` uses a non-static assert
- `FullPivLU.inverse()` uses no assert
- compilation still fails for large matrices, since `FullPivLU.inverse()` returns a different result type than expected